src/Control/Lens/Iso.hs

```diff
@@ -427,81 +409,25 @@ flipped :: Iso (a -> b -> c) (a' -> b' -> c') (b -> a -> c) (b' -> a' -> c')
 flipped = iso flip flip
 {-# INLINE flipped #-}
 
--- | This class provides for symmetric bifunctors.
-class Bifunctor p => Swapped p where
-  -- |
-  -- @
-  -- 'swapped' '.' 'swapped' ≡ 'id'
-  -- 'first' f '.' 'swapped' = 'swapped' '.' 'second' f
-  -- 'second' g '.' 'swapped' = 'swapped' '.' 'first' g
-  -- 'bimap' f g '.' 'swapped' = 'swapped' '.' 'bimap' g f
-  -- @
-  --
-  -- >>> (1,2)^.swapped
-  -- (2,1)
-  swapped :: Iso (p a b) (p c d) (p b a) (p d c)
-
-instance Swapped (,) where
-  swapped = iso swap swap
-
-instance Swapped Either where
-  swapped = iso (either Right Left) (either Right Left)
-
-instance (Swapped f, Swapped g) => Swapped (Product f g) where
-  swapped = iso f f
-    where
-      f (Pair x y) = Pair (x ^. swapped) (y ^. swapped)
-
-instance (Swapped p, Swapped q) => Swapped (Sum p q) where
-  swapped = iso f f
-    where
-      f (L2 x) = L2 (x ^. swapped)
-      f (R2 x) = R2 (x ^. swapped)
-
-instance (Swapped p) => Swapped (Flip p) where
-  swapped = iso f f
-    where
-      f (Flip p) = Flip (p ^. swapped)
-
-instance (f ~ g, Functor f, Swapped p) => Swapped (Biff p f g) where
-  swapped = iso f f
-    where
-      f (Biff p) = Biff (p ^. swapped)
-
-instance (Functor f, Swapped p) => Swapped (Tannen f p) where
-  swapped = iso f f
-    where
-      f (Tannen x) = Tannen $ fmap (^. swapped) x
-
-instance Swapped ((,,) x) where
-  swapped = iso f f
-    where
-      f (x,a,b) = (x,b,a)
-
-instance Swapped ((,,,) x y) where
-  swapped = iso f f
-    where
-      f (x,y,a,b) = (x,y,b,a)
-
-instance Swapped ((,,,,) x y z) where
-  swapped = iso f f
-    where
-      f (x,y,z,a,b) = (x,y,z,b,a)
-
-instance Swapped ((,,,,,) x y z w) where
-  swapped = iso f f
-    where
-      f (x,y,z,w,a,b) = (x,y,z,w,b,a)
-
-instance Swapped ((,,,,,,) x y z w v) where
-  swapped = iso f f
-    where
-      f (x,y,z,w,v,a,b) = (x,y,z,w,v,b,a)
+-- |
+-- @
+-- 'swapped' '.' 'swapped' ≡ 'id'
+-- 'first' f '.' 'swapped' = 'swapped' '.' 'second' f
+-- 'second' g '.' 'swapped' = 'swapped' '.' 'first' g
+-- 'bimap' f g '.' 'swapped' = 'swapped' '.' 'bimap' g f
+-- @
+--
+-- >>> (1,2)^.swapped
+-- (2,1)
+swapped :: Swap p => Iso (p a b) (p c d) (p b a) (p d c)
+swapped = iso swap swap
+{-# INLINE swapped #-}
 
 -- | Ad hoc conversion between \"strict\" and \"lazy\" versions of a structure,
```

Comment:
These Haddocks previously described the `Strict` class, but they now feel a bit off when describing the `strict` function, which is a combinator built on top of `Strict`. This may come down to personal taste, but I would prefer making the Haddocks more closely resemble those of `lazy`, i.e.,

```hs
-- | An 'Iso' between the lazy variant of a structure and its strict
-- counterpart.
--
-- @
-- 'strict' = 'from' 'lazy'
-- @
```